File: removeblack.m
function [imageOut] = removeblack(imageIn, replaceValue, varargin)
%UNTITLED Summary of this function goes here
%   Detailed explanation goes here

%% input layer
if ~exist('replaceValue'), replaceValue = 256; end

%% detect black areas
imageGray = rgb2gray(imageIn);
maskBlack = imageGray <5;
maskBlack =repmat(maskBlack, [1,1,3]);

%% remove it 
imageOut = imageIn;
imageOut(maskBlack)= replaceValue;

end

File: python/removeblack.py

import numpy as np
from skimage.color import rgb2gray

def removeBlack(imageIn, replaceValue=255):
    
    # detect black areas
    imageGray = rgb2gray(imageIn)
    imageGray = np.uint8(rgb2gray(imageIn)*255)
    maskBlack = imageGray < 5

    # remove it
    imageIn[maskBlack] = replaceValue
    
    return imageIn